File: zoomx/src/main/java/com/zoomx/zoomx/ui/requestdetails/RequestDetailsViewModel.java
package com.zoomx.zoomx.ui.requestdetails;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.zoomx.zoomx.manager.ZoomX;
import com.zoomx.zoomx.model.RequestEntity;

/**
 * Created by Ibrahim AbdelGawad on 12/13/2017.
 */

public class RequestDetailsViewModel extends ViewModel {

    public LiveData<RequestEntity> getRequestById(int id) {
        return ZoomX.getRequestDao().getRequestById(id);
    }
}

File: zoomx/src/main/java/com/zoomx/zoomx/ui/requestdetails/RequestDetailsViewModel.kt
package com.zoomx.zoomx.ui.requestdetails

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel

import com.zoomx.zoomx.config.ZoomX
import com.zoomx.zoomx.model.RequestEntity

/**
 * Created by Ibrahim AbdelGawad on 12/13/2017.
 */

class RequestDetailsViewModel : ViewModel() {

    fun getRequestById(id: Int): LiveData<RequestEntity> {
        return ZoomX.requestDao!!.getRequestById(id)
    }
}
